cli/tests/activate/nested-activate-fish.exp

```diff
@@ -0,0 +1,40 @@
+# Activate a project environment using --dir
+
+set dir [lindex $argv 0]
+set flox $env(FLOX_BIN)
+set timeout 10
+
+spawn $flox activate --dir $dir/outer
+expect_after {
+  timeout { exit 1 }
+  eof { exit 2 }
+  "*\n" { exp_continue }
+  "*\r" { exp_continue }
+}
+expect "You are now using the environment"
+
+send "$flox activate --dir $dir/inner\r"
+expect "You are now using the environment"
```

Comment:
suggestion blocking: I think you probably need the whole `expect_after` construct from above. I would have to dig through git history to figure out why

Reply:
The `expect_after` is structured that way to deal with bootstrapping issues for the tty and login shell set up by expect itself, and by the point we get to the point of invoking the nested activation we don't need to fear that sort of failure. But fine, we can replace this `expect` with the same `expect_after` block, it's just unnecessary IMO.

Reply:
If you've checked we don't need it I don't feel we need to put it in, I just wasn't sure

Reply:
I think the relevant commit is 4d99a52a89761accb99554fdd1a08aea17e9f95f. We had some flakiness before that

Reply:
Ah, that's the commit where we lost the timeout message. 😢  Not a lot of explanation given in the commit comment for the analysis, options considered, etc. either, so I'm really not any more or less inclined to add it based on that. So ... I'll go ahead and add it to close out the conversation.

Reply:
commit 2507bc104cf8beef8b92b01b3bd732a27a9c2418